.make/test.mk

```diff
@@ -0,0 +1,546 @@
+#===============================================================================
+# Testing has become a rather big and interconnected topic and that's why it
+# has arrived in it's own file.
+#
+# We have to types of tests available:
+#
+#  1. unit tests
+#  2. integration tests and
+#  3. remote tests.
+#
+# While the unit tests can be executed fairly simply be running `go test`, the
+# integration tests have a little bit more setup going on. The remote tests reqire
+# availability of some remote servers such as Keycloak.
+# That's why they are split up in three tests.
+#
+# Usage
+# -----
+# If you want to run the unit tests, type
+#
+#     $ make test-unit
+#
+# To run the integration tests, type
+#
+#     $ make test-integration
+#
+# To run the remote tests, type
```

Comment:
I don't think we have remote tests

Reply:
yup, but I thought to keep all `test.mk` target as it is as addding/removing in future is error prone.  but I guess you have concern for that so I will remove all remote_test, coverage_tests, db_stuff from `test.mk` file.

Reply:
yeah, I believe it'll be easier to maintain if you remove all unnecessary stuff here ;)